### DISCUSSION TITLE
Request for accessibility considerations

### DISCUSSION THREAD

**Discussion by rpccl** [lichess-org/mobile#14] [General]
Lichess believes that everyone should be able to play chess regardless of the age, gender or disability. I am a visually challenged person. I think Lichess is the only major website which is accessible with screen readers. I would love to see mobile application which is accessible with screen readers for visually challenge people. Specially the chess board should be accessible. I should be able to click on a peace to select it and should be able to click a square to drop the peace there. moves should be announced by screen reader. I should be able to read Times of both players with screen reader. Thank you.

**Comment by Omrimg2** [comment]
Great idea. Luckily, Flutter's support for screen readers is pretty sweet using the [Semantics](https://api.flutter.dev/flutter/widgets/Semantics-class.html) class. [This](https://blog.gskinner.com/archives/2022/09/flutter-crafting-a-great-experience-for-screen-readers.html ) is a nice blog piece describing the process, just putting it out here.

**Comment by rpccl** [comment]
Thank you for considering my Idea. And yes I am aware that flutter is fully accessible with screen readers. When I wanted to create a chess app I was unable to find any chess board package which was accessible with screen readers. That is my concern. So if a custom package for chess board will be used throughout the app make sure its accessible. Otherwise please consider creating a package which is accessible for all. I will be giving my feedbacks how ever and wherever I can. Thank you.

**Comment by veloce** [comment]
I started to do it here: https://github.com/lichess-org/mobile/commit/bdadab6c124d0e8502235c5dc2333bab8ac9805e

Then plan is to make the app fully accessible, including all new features. Next step is to work on a way to make moves using a keyboard.

**Comment by rpccl** [comment]
First of all thank you so much for considering the request. having keyboard
support is very good but I would also like to have following
functionality.  I should be able to make moves just by clicking on source
square and target square. I appreciate each and every efforts which are
being made to improve accessibility. Thank you so much.
